drivers: sensor: tmp1075: Add tmp1075 sensor driver and sample

Paplewski opened the pull request:
TI tmp1075 driver implemented based on tmp108 driver. The driver initializes the sensor based on the DTS. See sample app.overlay.

Added sample to show example driver usage.
To see default DTS configuration option inspect `ti,tmp1075.yaml` bindings file.

Paplewski committed:
drivers: sensor: tmp1075: Add tmp1075 sensor driver and sample

TI tmp1075 driver implemented based on tmp108 driver.
The driver initializes the sensor based on the DTS.

Added tmp1075 example overlay file to thermometer sample.
All you need to do to use the sensor is to connect the I2C and
optionally interrupt line.
To see default DTS configuration option inspect `ti,tmp1075.yaml`
bindings file and sensor spec.

Signed-off-by: Paweł Czaplewski <pawel.czaplewski@arrow.com>

MaureenHelm commented on drivers/sensor/ti/tmp1075/tmp1075.c:
```diff
@@ -0,0 +1,254 @@
+	const struct gpio_dt_spec *alert_gpio = &config->alert_gpio;
+	int result;
+
+	if (!device_is_ready(alert_gpio->port)) {
```
Use `gpio_is_ready_dt`

Paplewski replied:
done

MaureenHelm commented on drivers/sensor/ti/tmp1075/tmp1075.c:
```diff
@@ -0,0 +1,254 @@
+}
+#endif
+
+int tmp1075_init(const struct device *dev)
```
```suggestion
static int tmp1075_init(const struct device *dev)
```

Paplewski replied:
updated

MaureenHelm commented on drivers/sensor/ti/tmp1075/tmp1075.c:
```diff
@@ -0,0 +1,254 @@
+	const struct tmp1075_config *cfg = dev->config;
+	struct tmp1075_data *data = dev->data;
+
+	if (!device_is_ready(cfg->bus.bus)) {
```
Use `i2c_is_ready_dt`

Paplewski replied:
done

MaureenHelm commented on drivers/sensor/ti/tmp1075/tmp1075_trigger.c:
```diff
@@ -0,0 +1,54 @@
+
+	if (trig->type == SENSOR_TRIG_THRESHOLD) {
+		drv_data->temp_alert_handler = handler;
+		drv_data->temp_alert_trigger = *trig;
```
Save the address of the trigger, not a copy of the trigger.

Paplewski replied:
done

MaureenHelm commented on dts/bindings/sensor/ti,tmp1075.yaml:
```diff
@@ -0,0 +1,55 @@
+  conversion-rate:
+    description: Conversion rate in ms.
+    type: int
+    default: 220
```
Need to explain in the description why the default value was selected
https://docs.zephyrproject.org/latest/build/dts/bindings-upstream.html#dt-bindings-default-rules

Paplewski replied:
removed default value and updated overlay file

MaureenHelm commented on dts/bindings/sensor/ti,tmp1075.yaml:
```diff
@@ -0,0 +1,55 @@
+    type: int
+    default: 220
+    enum:
+      - 275 # 27.5ms
```
This doesn't make sense

Paplewski replied:
removed default value and updated overlay file

Paplewski replied:
changed to microseconds

MaureenHelm commented on dts/bindings/sensor/ti,tmp1075.yaml:
```diff
@@ -0,0 +1,55 @@
+  consecutive-fault-measurements:
+    description: Number of consecutive measured faults that will trigger the alert.
+    type: int
+    default: 4
```
Need to explain in the description why the default value was selected
https://docs.zephyrproject.org/latest/build/dts/bindings-upstream.html#dt-bindings-default-rules

Paplewski replied:
removed default value and updated overlay file

MaureenHelm commented on dts/bindings/sensor/ti,tmp1075.yaml:
```diff
@@ -0,0 +1,55 @@
+  lower-threshold:
+    description: Lower threshold for alert interrupt. Expressed in degrees C.
+    type: int
+    default: 28
```
Need to explain in the description why the default value was selected
https://docs.zephyrproject.org/latest/build/dts/bindings-upstream.html#dt-bindings-default-rules

Paplewski replied:
removed default value and updated overlay file

MaureenHelm commented on dts/bindings/sensor/ti,tmp1075.yaml:
```diff
@@ -0,0 +1,55 @@
+  upper-threshold:
+    description: Upper threshold for alert interrupt. Expressed in degrees C.
+    type: int
+    default: 29
```
Need to explain in the description why the default value was selected
https://docs.zephyrproject.org/latest/build/dts/bindings-upstream.html#dt-bindings-default-rules

Paplewski replied:
removed default value and updated overlay file

MaureenHelm reviewed (changes requested):
Thanks for the contribution!

Please use the existing `samples/sensor/thermometer` instead of introducing a driver-specific sample.

MaureenHelm commented on tests/drivers/build_all/sensor/i2c.dtsi:
```diff
@@ -1113,3 +1113,14 @@ test_i2c_mmc56x3: mmc56x3@92 {
 	auto-self-reset;
 };
+
+test_i2c_tmp1075: tmp1075@91 {
```
The address should be 0x98. The previous node is incorrect and should be 0x97, but that can be fixed in a separate patch.

Paplewski replied:
updated both addresses in a separate commit

Paplewski committed:
tests: drivers: i2c: fix I2C registers configuration in i2c.dtsi

Fixed incorrect I2C register definitions in the test file
tests/drivers/build_all/sensor/i2c.dtsi.

Signed-off-by: Paweł Czaplewski <pawel.czaplewski@arrow.com>

ubieda commented on drivers/sensor/ti/tmp1075/Kconfig:
```diff
@@ -0,0 +1,24 @@
+
+if TMP1075
+
+config TMP1075_ALERT_INTERRUPTS
+	bool "Allow interrupts to service over and under temp alerts"
+	help
+	  This will set up interrupts to service under and over temp alerts
+	  see TMP1075 spec sheet for more information on how these work.
```
May we apply this pattern enforced in this (on-going) PR? #79226
